{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3801d333",
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "import os\n",
    "import re\n",
    "import subprocess\n",
    "import shutil\n",
    "import yaml\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2b7c76f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'C3L-00677',\n",
       " 'rna-seq': {'tumor': {'R1': {'filepath': '/data/pecgs/test_sample/C3L-00677/rna-seq/170906_UNC32-K00270_0058_BHLJ7FBBXX_TGACCA_S4_L007_R1_001.fastq.gz'},\n",
       "   'R2': {'filepath': '/data/pecgs/test_sample/C3L-00677/rna-seq/170906_UNC32-K00270_0058_BHLJ7FBBXX_TGACCA_S4_L007_R2_001.fastq.gz'}}},\n",
       " 'wxs': {'normal': {'barcode': 'CPT0071960002',\n",
       "   'filepath': '/data/pecgs/test_sample/C3L-00677/wxs/5cee818c-61e8-4441-bd9a-5c984cde9c6b_gdc_realn.bam'},\n",
       "  'tumor': {'barcode': 'CPT0071100006',\n",
       "   'filepath': '/data/pecgs/test_sample/C3L-00677/wxs/1d6c4f0f-09c4-4472-a19e-30ee7e0ecf64_gdc_realn.bam'}}}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "m = yaml.safe_load(open('/data/pecgs/test_sample/C3L-00677/summary.yaml'))\n",
    "m"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50fbf538",
   "metadata": {},
   "source": [
    "###### top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "25e1ba92",
   "metadata": {},
   "outputs": [],
   "source": [
    "GENOME = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa'\n",
    "GENOME_DICT = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.dict'\n",
    "TARGET_INTERVAL_LIST = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/cnv/db/IDT_xGen_Exome_Research_Panel_v1.hg38.removed_alt_chr.bed.target.preprocessed.exome.interval_list'\n",
    "COMMON_BIALLELIC = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/cnv/db/snp_database/af-only-gnomad.hg38.common_biallelic.chr1-22XY.vcf'\n",
    "PROTEIN_CODING_GENE = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/cnv/db/remaking_protein-coding/gencode.v34.annotation.gene_filterd.need_gene_symbol.no_sym.filtered_to_hgnc_protein-coding_genes.bed'\n",
    "POOL_OF_NORMALS = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/cnv/pon/create_pon/gatk4scnaPON.Normal.hdf5'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "688417d5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e7f38909",
   "metadata": {},
   "source": [
    "###### step 3: STAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fc5a55a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # sh $submitJob 16 1 gatk4cn.s3.${caseID} \"bash $scriptDir/src/3.run.gatk4scna.callCNA.pair.sh -C ${config} -P ${CNVPON} -S ${caseID} -N ${normalBam} -T ${tumorBam} -O ${outdir}\"\n",
    "\n",
    "\n",
    "# ##---------- TotalCNA Tumor BAM\n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx16g -jar ${GATK} CollectReadCounts \\\n",
    "#       -I ${BAM_Tumor} \\\n",
    "#       -L ${TargetIntervalList} \\\n",
    "#       --interval-merging-rule OVERLAPPING_ONLY \\\n",
    "#       -O ${OUT}/${caseName}.T.counts.hdf5\n",
    "\n",
    "\n",
    "\n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx16g -jar ${GATK} DenoiseReadCounts \\\n",
    "#       -I ${OUT}/${caseName}.T.counts.hdf5 \\\n",
    "#       --count-panel-of-normals ${CNVPON} \\\n",
    "#       --standardized-copy-ratios ${OUT}/${caseName}.T.standardizedCR.tsv \\\n",
    "#       --denoised-copy-ratios ${OUT}/${caseName}.T.denoisedCR.tsv\n",
    "\n",
    "\n",
    "# # Need large memory\n",
    "# ##---------- AlleleCNA Normal BAMs\n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx16g -jar ${GATK} CollectAllelicCounts \\\n",
    "#       -L ${COMMON_BIALLELIC} \\\n",
    "#       -I ${BAM_Normal} \\\n",
    "#       -R ${GENOME} \\\n",
    "#       -O ${OUT}/${caseName}.N.allelicCounts.tsv\n",
    "\n",
    "\n",
    "\n",
    "# ##---------- AlleleCNA Tumor BAMs\n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx16g -jar ${GATK} CollectAllelicCounts \\\n",
    "#       -L ${COMMON_BIALLELIC} \\\n",
    "#       -I ${BAM_Tumor} \\\n",
    "#       -R ${GENOME} \\\n",
    "#       -O ${OUT}/${caseName}.T.allelicCounts.tsv\n",
    "\n",
    "      \n",
    "\n",
    "# ##---------- AlleleCNA Nor & Tum \n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx16g -jar ${GATK} ModelSegments \\\n",
    "#       --denoised-copy-ratios ${OUT}/${caseName}.T.denoisedCR.tsv \\\n",
    "#       --allelic-counts ${OUT}/${caseName}.T.allelicCounts.tsv \\\n",
    "#       --normal-allelic-counts ${OUT}/${caseName}.N.allelicCounts.tsv \\\n",
    "#       --output ${OUT} \\\n",
    "#       --output-prefix ${caseName}.T\n",
    "    \n",
    "\n",
    "    \n",
    "# ##---------- Segment \n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx16g -jar ${GATK} CallCopyRatioSegments \\\n",
    "#       --input ${OUT}/${caseName}.T.cr.seg \\\n",
    "#       --output ${OUT}/${caseName}.T.called.seg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2791787a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ##---------- Plot-TotalCN\n",
    "# ${JAVA} -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -Xmx8g -jar ${GATK} PlotDenoisedCopyRatios \\\n",
    "#       --standardized-copy-ratios ${OUT}/${NAME}.standardizedCR.tsv \\\n",
    "#       --denoised-copy-ratios ${OUT}/${NAME}.denoisedCR.tsv \\\n",
    "#       --sequence-dictionary ${GENOME_DICT} \\\n",
    "#       --minimum-contig-length ${min_contigLen} \\\n",
    "#       --output ${OUT}/plots \\\n",
    "#       --output-prefix ${NAME} \\\n",
    "#       --tmp-dir ${OUT}\n",
    "\n",
    "\n",
    "\n",
    "# ##---------- Plot-alleleCN\n",
    "# ${JAVA}  -Dsamjdk.use_async_io_read_samtools=false \\\n",
    "#       -Dsamjdk.use_async_io_write_samtools=true \\\n",
    "#       -Dsamjdk.use_async_io_write_tribble=false \\\n",
    "#       -Dsamjdk.compression_level=2 \\\n",
    "#       -jar ${GATK} PlotModeledSegments \\\n",
    "#       --denoised-copy-ratios ${OUT}/${caseName}.T.denoisedCR.tsv \\\n",
    "#       --allelic-counts ${OUT}/${caseName}.T.hets.tsv \\\n",
    "#       --segments ${OUT}/${caseName}.T.modelFinal.seg \\\n",
    "#       --sequence-dictionary ${GENOME_DICT} \\\n",
    "#       --minimum-contig-length ${min_contigLen} \\\n",
    "#       --output ${OUT}/plots \\\n",
    "#       --output-prefix ${caseName}.T \\\n",
    "#       --tmp-dir ${OUT}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d58cd838",
   "metadata": {},
   "outputs": [],
   "source": [
    "def collect_read_counts(tumor_bam, sample, out_dir):\n",
    "    pieces = [\n",
    "        'gatk CollectReadCounts',\n",
    "        '-I', tumor_bam,\n",
    "        '-L', TARGET_INTERVAL_LIST,\n",
    "        '--interval-merging-rule OVERLAPPING_ONLY',\n",
    "        '-O', os.path.join(out_dir, f'{sample}.T.counts.hdf5')\n",
    "    ]\n",
    "    return ' '.join(pieces)\n",
    "\n",
    "\n",
    "def denoise_counts(sample, out_dir):\n",
    "    pieces = [\n",
    "        'gatk DenoiseReadCounts',\n",
    "        '-I', os.path.join(out_dir, f'{sample}.T.counts.hdf5'),\n",
    "        '--count-panel-of-normals', POOL_OF_NORMALS,\n",
    "        '--standardized-copy-ratios', os.path.join(out_dir, f'{sample}.T.standardizedCR.tsv'),\n",
    "        '--denoised-copy-ratios', os.path.join(out_dir, f'{sample}.T.denoisedCR.tsv')\n",
    "    ]\n",
    "    return ' '.join(pieces)\n",
    "\n",
    "\n",
    "def collect_allelic_counts(bam, sample, out_dir, char):\n",
    "    pieces = [\n",
    "        'gatk CollectAllelicCounts',\n",
    "        '-L', COMMON_BIALLELIC,\n",
    "        '-I', bam,\n",
    "        '-R', GENOME,\n",
    "        '-O', os.path.join(out_dir, f'{sample}.{char}.allelicCounts.tsv')\n",
    "    ]\n",
    "    return ' '.join(pieces)\n",
    "\n",
    "\n",
    "def model_segments(sample, out_dir):\n",
    "    pieces = [\n",
    "        'gatk ModelSegments',\n",
    "        '--denoised-copy-ratios', os.path.join(out_dir, f'{sample}.T.denoisedCR.tsv'),\n",
    "        '--allelic-counts', os.path.join(out_dir, f'{sample}.T.allelicCounts.tsv'),\n",
    "        '--normal-allelic-counts', os.path.join(out_dir, f'{sample}.N.allelicCounts.tsv'),\n",
    "        '--output', out_dir,\n",
    "        '--output-prefix', f'{sample}.T'\n",
    "    ]\n",
    "    return ' '.join(pieces)\n",
    "\n",
    "\n",
    "def call_copy_segments(sample, out_dir):\n",
    "    pieces = [\n",
    "        'gatk CallCopyRatioSegments',\n",
    "        '--input', os.path.join(out_dir, f'{sample}.T.cr.seg'),\n",
    "        '--output', os.path.join(out_dir, f'{sample}.T.called.seg')\n",
    "    ]\n",
    "    return ' '.join(pieces)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c3952590",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_denoised(sample, out_dir, min_contig_len=10000000, char='T'):\n",
    "    pieces = [\n",
    "        'gatk PlotDenoisedCopyRatios',\n",
    "        '--standardized-copy-ratios', os.path.join(out_dir, f'{sample}.{char}.standardizedCR.tsv'),\n",
    "        '--denoised-copy-ratios', os.path.join(out_dir, f'{sample}.{char}.denoisedCR.tsv'),\n",
    "        '--sequence-dictionary', GENOME_DICT,\n",
    "        '--minimum-contig-length', str(min_contig_len),\n",
    "        '--output', os.path.join(out_dir, 'plots'),\n",
    "        '--output-prefix', f'{sample}.{char}',\n",
    "        '--tmp-dir', out_dir\n",
    "    ]\n",
    "    return ' '.join(pieces)\n",
    "\n",
    "\n",
    "def plot_segments(sample, out_dir, min_contig_len=10000000, char='T'):\n",
    "    pieces = [\n",
    "        'gatk PlotModeledSegments',\n",
    "        '--denoised-copy-ratios', os.path.join(out_dir, f'{sample}.{char}.denoisedCR.tsv'),\n",
    "        '--allelic-counts', os.path.join(out_dir, f'{sample}.{char}.hets.tsv'),\n",
    "        '--segments', os.path.join(out_dir, f'{sample}.{char}.modelFinal.seg'),\n",
    "        '--sequence-dictionary', GENOME_DICT,\n",
    "        '--minimum-contig-length', str(min_contig_len),\n",
    "        '--output', os.path.join(out_dir, 'plots'),\n",
    "        '--output-prefix', f'{sample}.{char}',\n",
    "        '--tmp-dir', out_dir\n",
    "    ]\n",
    "    return ' '.join(pieces)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4671a2d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sh $submitJob 1 1 gatk4cn.s5.${caseID} \"${PYTHON3} $scriptDir/src/segment_to_geneLevel.py --prefix ${caseID}.T --name ${caseID} --seg ${DIR}/${caseID}.T.called.igv.seg --gene ${ProteinCodingGene} -o ${DIR}\"\n",
    "# sh $submitJob 2 1 gatk4cn.s6.${caseID} \"${PYTHON3} $scriptDir/src/mergeMultipleFilesToOne.py ${outdir}\"\n",
    "\n",
    "def gene_level(script, sample, out_dir):\n",
    "    pieces = [\n",
    "        'python', script,\n",
    "        '--prefix', f'{sample}.T',\n",
    "        '--name', sample,\n",
    "        '--seg', os.path.join(out_dir, f'{sample}.T.called.igv.seg'),\n",
    "        '--gene', PROTEIN_CODING_GENE,\n",
    "        '-o', out_dir\n",
    "    ]\n",
    "    return ' '.join(pieces)\n",
    "\n",
    "def merge_gene_level(script, out_dir):\n",
    "    pieces = [\n",
    "        'python', script, out_dir\n",
    "    ]\n",
    "    return ' '.join(pieces)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a5b42bbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_cnv(normal_bam, tumor_bam, sample, out_dir, gene_level_script, merge_gene_script):\n",
    "    logging.info('collecting tumor reads')\n",
    "    cmd = collect_read_counts(tumor_bam, sample, out_dir)\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('denoising counts')\n",
    "    cmd = denoise_counts(sample, out_dir)\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('collect allelic counts normal')\n",
    "    cmd = collect_allelic_counts(normal_bam, sample, out_dir, 'N')\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('collect allelic counts tumor')\n",
    "    cmd = collect_allelic_counts(tumor_bam, sample, out_dir, 'T')\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('modeling segments')\n",
    "    cmd = model_segments(sample, out_dir)\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('call copy ratio segments')\n",
    "    cmd = call_copy_segments(sample, out_dir)\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('plot denoised tumor')\n",
    "    cmd = plot_denoised(sample, out_dir, char='T')\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('plot segments tumor')\n",
    "    cmd = plot_segments(sample, out_dir, char='T')\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('plot denoised normal')\n",
    "    cmd = plot_denoised(sample, out_dir, char='N')\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('plot segments normal')\n",
    "    cmd = plot_segments(sample, out_dir, char='N')\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('running gene level')\n",
    "    cmd = gene_level(gene_level_script, sample, out_dir)\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n",
    "    logging.info('merging gene level')\n",
    "    cmd = merge_gene_level(merge_gene_script, out_dir)\n",
    "    logging.info(f'executing command: {cmd}')\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d8dff2c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "880d1a21",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aaa8d34a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49a945fc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
